### DISCUSSION TITLE
Whether you can support manual custom validation of DNS records in golang dependent mode?

### DISCUSSION THREAD

**Discussion by laixiaohuai** [go-acme/lego#2186] [Q&A]
### Welcome

- [X] Yes, I've searched similar issues on GitHub and didn't find any.

### How do you use lego?

Library

### Detailed Description

Whether you can support manual custom validation of DNS records in golang dependent mode?

**Comment by ldez** [comment]
Hello,

I don't understand your question, can you explain it?

**Reply by orange-juzipi** [reply]
What he meant should be to customize the records of domain name resolution, rather than automatically created and verified `FQDN` and `Value`

> info := dns01.GetChallengeInfo(domain, keyAuth)

Currently, the official `DNSProvider Presents` are automatically created to resolve and cannot be customized unless they are re-implemented by themselves.
```

// Present creates a TXT record to fulfill the dns-01 challenge.
func (d *DNSProvider) Present(domain, token, keyAuth string) error {
	info := dns01.GetChallengeInfo(domain, keyAuth)

	zoneName, err := d.getHostedZone(info.EffectiveFQDN)
	if err != nil {
		return fmt.Errorf("alicloud: %w", err)
	}

	recordAttributes, err := d.newTxtRecord(zoneName, info.EffectiveFQDN, info.Value)
	if err != nil {
		return err
	}

	_, err = d.client.AddDomainRecord(recordAttributes)
	if err != nil {
		return fmt.Errorf("alicloud: API call failed: %w", err)
	}
	return nil
}

```

**Reply by ldez** [reply]
You can use CNAMEs

**Reply by orange-juzipi** [reply]
Do you have an example of how to use CNAME?

**Reply by ldez** [reply]
https://letsencrypt.org/2019/10/09/onboarding-your-customers-with-lets-encrypt-and-acme.html#the-advantages-of-a-cname
